internal/apiresource/role.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright IBM Corporation 2020
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiresource
+
+import (
+	common "github.com/konveyor/move2kube/internal/common"
+	irtypes "github.com/konveyor/move2kube/internal/types"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	roleKind = "Role"
+)
+
+// Role handles all objects like a role.
+type Role struct {
+}
+
+// GetSupportedKinds returns the kinds that this type supports.
+func (*Role) GetSupportedKinds() []string {
+	return []string{roleKind}
+}
+
+// CreateNewResources creates the runtime objects from the intermediate representation.
+func (r *Role) CreateNewResources(ir irtypes.IR, supportedKinds []string) []runtime.Object {
+	irresources := ir.Roles
```

Comment:
check if role is supported

Reply:
if it is not supported what should happen? I am not sure there is a fallback.
So error and exit without creating the CI/CD pipeline?